mirdata/datasets/haydn_op20.py

```diff
@@ -0,0 +1,328 @@
+            index,
+            metadata,
+        )
+        self.humdrum_annotated_path = os.path.join(self._data_home, self._track_paths["annotations"][0])
+        self.title = os.path.splitext(self._track_paths["annotations"][0])[0]
+
+    def show(self):
+        show_score(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def score(self) -> music21.stream.Score:
+        return load_score(self.humdrum_annotated_path)
+
+    @core.cached_property
+    def keys(self, resolution=28) -> List[dict]:
+        return load_key(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def roman_numerals(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def chords(self, resolution=28) -> List[dict]:
+        return load_roman_numerals(self.humdrum_annotated_path, resolution)
+
+    @core.cached_property
+    def duration(self) -> int:
+        return self.chords[-1]['time']
+
+    @core.cached_property
+    def midi_path(self) -> str:
+        return load_midi_path(self.humdrum_annotated_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            metadata={
+                "duration": self.duration,
+                "title": self.title,
+                "keys": self.keys,
+                "roman_numerals": self.roman_numerals,
+                "midi_path": self.midi_path,
+                "score": self.score,
+                "humdrum_annotated_path": self.humdrum_annotated_path
+            },
+        )
+
+
+def load_score(path):
+    """Load haydn op20 score from a file with music21 format (music21.stream.Score).
+
+            Args:
+                path: path to hrm annotations
+
+            Returns:
+                music21.stream.Score: score in music21 format
+
+    """
+    score = music21.converter.parse(path, format='humdrum')
+    return score
+
+
+def show_score(path):
+    """Show score in Musescore 3. Required to have Musescore 3 installed.
+
+            Args:
+                path: path to hrm annotations
+
+
+    """
+    score = music21.converter.parse(path, format='humdrum')
+    score.show()
+
+
+def load_key(path, resolution=28):
+    """Load haydn op20 key data from a file
+
+        Args:
+            path: path to hrm annotations
+
+        Returns:
+            List[dict]: musical key data and time
+
+    """
+    score = music21.converter.parse(path, format='humdrum')
+    rna = {rn.offset: rn for rn in list(score.flat.getElementsByClass('RomanNumeral'))}
+    annotations = []
+    for offset, rn in rna.items():
+        if not rn:
+            continue
+        time = int(round(float(offset * resolution)))
+        tonicizedKey = rn.secondaryRomanNumeralKey
+        key = tonicizedKey if tonicizedKey else rn.key
```

Comment:
Probably this is an identical behavior and less verbose? If that was my code from muspy, sorry I didn't catch it before.

```python
key = tonicizedKey or rn.key
```

Reply:
yes! It is from your Muspy code!

Reply:
I will change it!